Feature/IOPS-2995-Fixing EPS Feedback Issues

KazeemHamzat opened the pull request:
@omarisgreat: I raised this separate PR so we can track the EPS conversation on the previous PR. The issues raised by EPS have been resolved here with exception of the minimal task with no reference to resource. It is mandatory for an action to contain a sub-action or resource. So, here are the affected actions and the approach I have taken.
1.Prescription upload successful
To avoid duplicating the sub-action for Prescription Line Items(Medications), I have referenced a supposed "reference": "Task/Task-PrescriptionUpload-Example"
2. Nominated Release Request successful: As EPS said, this is only a status marker
I have referenced "reference": "Task/Task-PrescriptionRelease-Example"

KazeemHamzat committed:
Fixing Feedback EPS Feedback Issues

Fixing Feedback EPS Feedback Issues

omarisgreat commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.json:
```diff
@@ -0,0 +1,214 @@
+      }
+     ],
+     "resource": {
+      "reference": "Task/Task-PrescriptionUpload-Example"
```
Could this be a reference to the Message Bundle that created the Prescription instead? e.g. the one used in  EPS prescription-order?

omarisgreat commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.json:
```diff
@@ -0,0 +1,214 @@
+      }
+     ],
+     "resource": {
+      "reference": "Task/Task-PrescriptionRelease-Example"
```
Similarly, could this be a reference to the prescription-release response, which is a Parameters resource?

omarisgreat reviewed (dismissed):
Small changes to references in subactions

KazeemHamzat committed:
Added Bundle with Feedback changes

KazeemHamzat committed:
Update Xml bundle

Update Xml bundle

KazeemHamzat committed:
Delete JSon Bundle

Delete JSon Bundle

KazeemHamzat committed:
Create Parameter-MedicationNotificationRelease-Example

Create Parameter-MedicationNotificationRelease-Example

KazeemHamzat committed:
Remove Json RequestGroup

KazeemHamzat committed:
Rename Parameter

KazeemHamzat commented:
[MatthewPopat-NHS](https://github.com/MatthewPopat-NHS), we have created this PR containing only FHIR assets you have requested changes on. Can we ask for a review to ascertain if all issues have now been fixed?

MatthewPopat-NHS commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -64,39 +58,37 @@
+    <action>
      <title value="Prescription status transitions" />
-     <action id="PrescriptionStatus-MedicationNumber-1">
+     <timingDateTime value="2025-01-29T13:00:00Z" />
```
timing on the outer action doesn't make sense, it should replace the timingTiming below

MatthewPopat-NHS commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -64,39 +58,37 @@
      <title value="Prescription Line Items(Medications)" />
-     <action id="Medication-Number-1">
+     <timingTiming>
+      <event value="2025-01-29T12:00:00Z" />
```
This is duplicate of authoredOn do we need it / want it?

omarisgreat replied:
This item is specifically the repeat information for the prescription as per https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/455#discussion_r2097778411

MatthewPopat-NHS commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -125,46 +109,10 @@
       </participant>
       <resource>
-       <reference value="MedicationRequest/MedicationRequest-Prescription-Example" />
+       <reference value="Bundle/Bundle-RequestGroupPrescriptionDetails-Example" />
```
No point in referencing anything here

omarisgreat replied:
Unfortunately either a subaction or a reference is required by the core FHIR constraints on RequestGroup.

MatthewPopat-NHS replied:
I hadn't spotted that, that actually throws a spanner in the works for this whole strategy. Most of the actions that can happen to a prescription don't have a resource. We could just point them all at the RequestGroup or at the MedicationRequests or at dummy Tasks but all of those feel like hacks :/

KazeemHamzat replied:
We opted for a Task resource for reasons highlighted below.

- **RequestGroup**- Referencing the RequestGroup itself seem to be a circular reference.
- **MedicationRequest** - May create confusion. Likely to be misconstrued as a missing medication on the prescription

MatthewPopat-NHS commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -186,19 +133,18 @@
       </participant>
       <resource>
-       <reference value="MedicationRequest/MedicationRequest-Prescription-Example" />
+       <reference value="Parameter/Parameter-PrescriptionRelease-Example" />
```
This isn't something we can referenco, just leave this off

MatthewPopat-NHS commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -173,7 +121,6 @@
         <display value="With Dispenser" />
        </coding>
       </code>
-      <timingDateTime value="2025-01-29T13:00:00Z" />
```
This was correct, why have we removed it?

MatthewPopat-NHS commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -186,19 +133,18 @@
+        <display value="With Dispenser - Active" />
        </coding>
       </code>
-      <timingDateTime value="2025-01-30T10:00:00Z" />
```
This was correct, why have we removed it?

MatthewPopat-NHS commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -211,7 +157,7 @@
       </participant>
       <resource>
-       <reference value="MedicationRequest/MedicationRequest-RepeatPrescription-Example" />
+       <reference value="MedicationDispense/Medicationdispense-PendingCancellation-Example" />
```
In general a dispense notification would be made up of multiple MedicationDispense resources, we were proposing using a list of sub actions each of which referenced one of the MedicationDispenses.

MatthewPopat-NHS commented on MedicationRequest/MedicationRequest-Prescription-Example.xml:
```diff
@@ -82,4 +76,7 @@
  </substitution>
+ <eventHistory>
+  <reference value="Provenance/Provenace-OriginalPrescription-Example" />
+ </eventHistory>
```
where has this come from? we don't use MedicationRequest.eventHistory in EPS

MatthewPopat-NHS commented on Parameters/Parameter-PrescriptionRelease-Example.xml:
```diff
@@ -0,0 +1,35 @@
+<Parameters xmlns="http://hl7.org/fhir">
```
This isn't a resource we store, it's just the way of sending in parameters to the $release request in the EPS API. It doesn't make sense to exist here as a referenceable resource.

KazeemHamzat committed:
Update Feedback Issues

Update Feedback Issues

KazeemHamzat committed:
Remove Parameter-PrescriptionRelease-Example

Remove Parameter-PrescriptionRelease-Example

KazeemHamzat committed:
Update RequestGroup-MedicationsPrescription-Example.

Removing reference to parameter resource

KazeemHamzat committed:
Feedback Issues

Change Reference of RequestGroup.sub-action for Prescription upload to Task

vickyjaiswal0308 commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
Here are some of minor things which i noticed :                                                                                                                                 1. MedicationDispense references the authorizingPrescription as:
  reference value="MedicationRequest/MedicationRequest-RepeatPrescription-Example"                                                          But ensure consistency in reference casing — it's lowercase medicationrequest in  some of the places.                                           

2. Typo in "Organization" (currently "Organzation"). <organization>
 reference value="Organzation/Organization-DispensingChemist-Example"                                                             

3. Dispense Quantity Mismatch - MedicationDispense.quantity.value = 32
MedicationRequest.quantity.value = 30 (Dispenser has issued more than prescribed. Dispensers are not typically authorized to override the prescribed quantity without documented justification.)     
    

4.  Two MedicationRequest resources for the same medication (Paracetamol 500mg soluble tablets) exist:

MedicationRequest-Prescription-Example (active, item with dispenser)
MedicationRequest-RepeatPrescription-Example (also active, but flagged as “Pending Cancellation”) This can be confusing for dispensers as Both are "active", which might cause both to be dispensed despite one being “Pending Cancellation”.

KazeemHamzat replied:
> 3\. MedicationDispense.quantity.value

Feedback 1: medicationrequest in lower cases appears CodeSystems canonical URL. Thus, its correct. No changes required.
Feedback 2: Typo error corrected.
Feedback 3: MedicationDispense relates to MedicationRequest-PrescriptionNumber2-Example. Both now have 32.
Feedback 4: The examples context aligns with template example provded by EPS. The whole idea is to wrap existing MedicationRequest`s within a RequestGroup resource. So, we expect that valid MedicationRequest will be used.

Feedback 5: MedicationRequest-RepeatPrescription-Example name has been changed to MedicationRequest-PrescriptionNumber2-Example. Extension-PendingCancellation changed to True. So, this should resolve the concern about 'Acute' and 'active' status.

vickyjaiswal0308 commented on MedicationRequest/MedicationRequest-Prescription-Example.xml:
1. Mismatch Between dosageInstruction.text and Dosage Form (Issue - The dosage instruction refers to a capsule, but the prescribed item is soluble tablets.)Recommendation: Change dosageInstruction.text to: "Take one soluble tablet three times a day"
2. authoredOn is required in many systems to trace the creation date and requester (e.g., GP or prescriber) is missing, which is vital for audit trail and accountability.
3. No validityPeriod in dispenseRequest - The absence of a validity window means this prescription may be filled indefinitely depending on the pharmacy system.

KazeemHamzat replied:
1 and 3: See Response for Feedback 4 and 5
2. authoredOn is not mandatory on both base R4 MedicationRequest, NHSDigital-MedicationRequest and EPS Template example followed as guidance.

vickyjaiswal0308 commented on RequestGroup/RequestGroup-MedicationsPrescription-Example.xml:
1. acute Course of Therapy on Repeat Prescription. It is marked as as a repeat prescription, yet declared it an acute therapy type. Acute therapies are non-repeatable by nature. If this is a repeat item then we can change it to code value="continuous" 
display value="Continuous long term therapy" 
2. Dispense Status Set to Item not dispensed in Active Prescription. Status indicates the item was not dispensed, but the MedicationRequest status is active.
3. authoredOn is required in many systems to trace the creation date and requester (e.g., GP or prescriber) is missing, which is vital for audit trail and accountability.

KazeemHamzat replied:
1 and 2: See Response for Feedback 4 and 5
3. authoredOn is not mandatory on both base R4 MedicationRequest, NHSDigital-MedicationRequest and EPS Template example followed as guidance.

KazeemHamzat committed:
Feedback Issues

Feedback 1: medicationrequest in lower cases appears CodeSystems canonical URL. Thus, its correct. No changes required.
Feedback 2: Typo error corrected.
Feedback 3: MedicationDispense relates to MedicationRequest-PrescriptionNumber2-Example. Both now have 32.
Feedback 4: The examples context aligns with template example provded by EPS. The whole idea is to wrap existing MedicationRequest`s within a RequestGroup resource. So, we expect that valid MedicationRequest will be used.

Feedback 5: MedicationRequest-RepeatPrescription-Example name has been changed to MedicationRequest-PrescriptionNumber2-Example. Extension-PendingCancellation changed to True. So, this should resolve the concern about 'Acute' and 'active' status.

KazeemHamzat committed:
Feedback Issues

Feedback Issues

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

KazeemHamzat committed:
Update NHSDigital-MedicationDispense

Update NHSDigital-MedicationDispense

KazeemHamzat committed:
Update MedicationRequest-PrescriptionNumber2-Example

Update MedicationRequest-PrescriptionNumber2-Example

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

vickyjaiswal0308 commented:
Raised review feedback seems resolved now but i can see some validation errors.

KazeemHamzat commented:
I have fixed the feedback issues raised by @vickyjaiswal0308 but now getting a weird error that seem to a Validator issue for the file:[Bundle-RequestGroupPrescriptionDetails-Example.xml](https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/blob/TestValidator/Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml). This error does not come up on other validators. HL7, Simplifier and Swagger version.

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Remove Meta.profile from PractitionerRole

KazeemHamzat commented on Bundle/Bundle-RequestGroupPrescriptionDetails-Example.xml:
```diff
@@ -221,6 +169,47 @@
   </search>
  </entry>
+ <entry>
+  <fullUrl value="http://example.org/fhir/PractitionerRole/PractitionerRole-JanePittman-Example" />
```
@vickyjaiswal0308, here is the Bundle entry for PractitionerRole/PractitionerRole-JanePittman-Example. Can you validate the Bundle using validators. HL7, Simplifier and Swagger version. It will pass in all of them. But I will check once more tomorrow before quitting.

vickyjaiswal0308 replied:
Yes, i can see it's passing in other validators . But can you try including profile like below and then see if the error is still occuring or not : 
`<PractitionerRole xmlns="http://hl7.org/fhir">
  <id value="PractitionerRole-JanePittman-Example"/>
  <meta>
    <profile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-PractitionerRole-SDS"/>
  </meta>
  ...
</PractitionerRole>`

vickyjaiswal0308 replied:
If it'll still occur . I'll approve it and then you can merge it into the main branch.

KazeemHamzat replied:
Victory at last. Now passing validation. The error relates Bundle entry 7. Organization referenced within PractitionerRole-JanePittman.

KazeemHamzat committed:
Add Meta Profile

Add Meta Profile

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example

KazeemHamzat committed:
Update CapabilityStatement-nhsdigital

Update CapabilityStatement-nhsdigital

KazeemHamzat committed:
Removing Meta Profile

Removing Meta Profile

KazeemHamzat committed:
Update Bundle-RequestGroupPrescriptionDetails-Example

Update Bundle-RequestGroupPrescriptionDetails-Example
